lib/puppet/application/master.rb

```diff
@@ -295,6 +295,9 @@ def start_webrick_master
     daemon.server = Puppet::Network::Server.new(Puppet[:bindaddress], Puppet[:masterport])
     daemon.daemonize if Puppet[:daemonize]
 
+    # Setup signal traps immediately after daemonization so we clean up the daemon
+    daemon.set_signal_traps
```

Comment:
Seems like it, but I'm not sure how that would interact with the `fork`. Potentially we could still SIGINT/SIGTERM between daemonize and setting up signal traps.